tests/functional/test_api.py

```diff
@@ -6,73 +6,46 @@
 from ghmirror.app import APP
 from ghmirror.data_structures.monostate import UsersCache
 
+class MockResponse:
+    def __init__(self, content, headers, status_code, user):
+        self.content = content.encode()
+        self.headers = headers
+        self.status_code = status_code
+        self.user = user
 
-def mocked_requests_get_etag(*args, **kwargs):
-    class MockResponse:
-        def __init__(self, content, headers, status_code):
-            self.content = content.encode()
-            self.headers = headers
-            self.status_code = status_code
+    def json(self):
+        return {'login': self.user}
```

Comment:
Nice work here! I know that was not the reason, but the deduplication is much appreciated.